language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java

```diff
@@ -424,6 +427,12 @@ public void visit(WhileStatementNode whileStatementNode) {
 
     @Override
     public void visit(ConditionalExpressionNode conditionalExpressionNode) {
+        if (this.functionCallExpressionNode != null &&
+                PositionUtil.isWithinLineRange(this.functionCallExpressionNode.lineRange(),
+                conditionalExpressionNode.lhsExpression().lineRange())) {
+            checkAndSetTypeDescResult(TypeDescKind.BOOLEAN);
```

Comment:
Now that we have introduced the TypesAPI, we can use `checkAndSetTypeResult(semanticModel.get().types().BOOLEAN)`